### ISSUE TITLE
SFTP downloads create un-deletable  duplicates of files

### ISSUE THREAD

**Issue by wyntrr-end** [isontheline/pro.webssh.net#392]
**Describe the bug**
When downloading files over SFTP to my device, I am prompted to choose where to save the file. The file then takes up space in that location, but the downloaded data also seems to be kept in WebSSH’s storage.
Example: Downloading a 4 GB video file and saving it to a folder in the “On My iPad” tab of Files saves the video in that location as expected. However, the size of the WebSSH app on my device is increased by ~4 GB as well. Deleting the video file from the Files app (and clearing “recently deleted”) frees up that 4 GB of space, but WebSSH still takes up an additional 4GB of space which can only be reclaimed by deleting the WebSSH app and re-downloading it from the App Store.

**Informations**
 - Device : iPad Mini 5
 - OS : iPad OS 14.7.1
 - WebSSH Pro version : 15.1.2

**Screenshots**
Fresh install of  WebSSH:
![IMG_2501](https://user-images.githubusercontent.com/66278138/131587145-bef52071-743c-417a-9545-59d49bfe3d14.png)
Saving file to my iPad via SFTP:
![IMG_2502](https://user-images.githubusercontent.com/66278138/131587172-7e88cf3f-2c04-48b6-9c35-b9691af06e35.jpeg)
File is saved in correct spot, as expected:
![IMG_2504](https://user-images.githubusercontent.com/66278138/131587231-26cb3976-88ec-4f32-816f-cdc57e9d2424.jpeg)
WebSSH now takes up a lot more space:
![IMG_2503](https://user-images.githubusercontent.com/66278138/131587263-251ff770-d139-4838-ad77-fd84b66b2d12.png)
Even after the file has been deleted from “On My iPad” and from “Recently Deleted”, WebSSH still takes up a lot of space:
![IMG_2505](https://user-images.githubusercontent.com/66278138/131587318-06e73a2e-0344-400e-9d30-539723c95a8c.png)

**To Reproduce**
1. Open system storage and take note of how much space is taken by WebSSH “Documents & Data”
2. Launch WebSSH and connect to a server via SFTP
3. Tap on a file and select “Download”
4. Once download is complete and you are prompted to choose a location to save it, do so
5. Return to system storage and see that WebSSH “Documents & Data” has increased by roughly the size of the file downloaded
6. Check that the selected download location also contains the file and takes up the expected amount of space
7. Delete the downloaded file using the “Files” app
8. Storage is reclaimed from the deleted file, but checking system storage shows that WebSSH still takes up additional space

**Comment by isontheline** [comment]
Hello @wyntrr-end 👋

Thank you a lot for these reproduce steps 🙏

I will fix this issue with a high priority.

Have a nice day ☀️

**Comment by isontheline** [comment]
Hello @wyntrr-end 

In the upcoming 16.0.2 Beta Release, temporary files will be deleted :
* at WebSSH startup
* at SFTP disconnection

**Comment by wyntrr-end** [comment]
@isontheline Awesome, thank you so much! I greatly appreciate your timely response

**Comment by isontheline** [comment]
Should be fixed in WebSSH 16.0.3 : https://testflight.apple.com/join/QSrBK59z
